lib/matplotlib/backend_bases.py

```diff
@@ -2977,6 +2977,66 @@ def _zoom_pan_handler(self, event):
             elif event.name == "button_release_event":
                 self.release_zoom(event)
 
+    def _start_event_axes_pan(self, event):
+        # call "ax.start_pan(..)" on all relevant axes of an event
+
+        def _ax_filter(ax):
+            return ax.in_axes(event) and ax.get_navigate() and ax.can_zoom()
+
+        def _capture_events(ax):
+            f = ax.get_forward_navigation_events()
+            if f == "auto":  # (capture = patch visibility)
+                f = not ax.patch.get_visible()
+            return not f
+
+        # get all relevant axes for the event
+        axes = list(filter(_ax_filter, self.canvas.figure.get_axes()))
+
+        if len(axes) == 0:
+            return
+
+        if self._nav_stack() is None:
+            self.push_current()   # Set the home button to this view.
+
+        # group axes by zorder (reverse to trigger later axes first)
+        grps = dict()
+        for ax in reversed(axes):
+            grps.setdefault(ax.get_zorder(), []).append(ax)
+
+        pan_axes = []
+        # go through zorders in reverse until we hit a capturing axes
+        for zorder in sorted(grps, reverse=True):
+            for ax in grps[zorder]:
+                pan_axes.append(ax)
+
+                # NOTE: shared axes are automatically triggered, but twin-axes
+                # and twin-axes of shared axes must be explicitly triggered!
+                pan_axes.extend(ax._twinned_axes.get_siblings(ax))
+
+                shared_axes = itertools.chain(
+                    *(ax._shared_axes[name].get_siblings(ax)
+                      for name in ax._axis_names))
+
+                for ax_s in shared_axes:
+                    ax_t = ax_s._twinned_axes.get_siblings(ax_s)
+                    # only trigger twinned-axes not the shared axes!
+                    pan_axes.extend((i for i in ax_t if i is not ax_s))
+
+                if _capture_events(ax):
+                    break  # break if we hit a capturing axes
+            else:
+                continue   # continue to lower zorder if we did not break
+
+            break          # break if we hit a capturing axes
```

Comment:
This is a dangling break?

Or now that I'm staring at this a bit more, does the `continue` in the `else:` clause above actually continue the outer for-loop? Maybe I'm just not picking up on it immediately, but the for-else with a continue not applying to that for-loop could use an extra comment here I think just to clarify what is explicitly going on.

Reply:
I understand that this is a rarely encountered construct... but in my opinion it's the easiest way to conditionally break out of nested loops...   I'll try to add more detailed comments to avoid confusion!

To clarify:  You are right, the `else: continue` is there to continue the outer loop only if no `break` was encountered in the inner loop. 

This way, if the inner loop breaks, the outer loop breaks as well, and if the inner loop finishes without an explicit break, the outer loop continues to the next level.

From the [docs](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops): 
> *When used with a loop, the else clause has more in common with the else clause of a [try](https://docs.python.org/3/reference/compound_stmts.html#try) statement than it does with that of [if](https://docs.python.org/3/reference/compound_stmts.html#if) statements: a [try](https://docs.python.org/3/reference/compound_stmts.html#try) statement’s else clause runs when no exception occurs, and a loop’s else clause runs when no break occurs.*

Reply:
I added a few more comments... please re-open this discussion in case you think my explanations are insufficient!